src/screens/ChamberOfSecrets.tsx

```diff
@@ -0,0 +1,222 @@
+/* eslint-disable react-native/no-raw-text */
+/* eslint-disable react-native/no-color-literals */
+/* eslint-disable react/no-unescaped-entities */
+import React, { useState } from 'react';
+import { StyleSheet, View } from 'react-native';
+import { connect } from 'react-redux';
+
+import {
+  Button,
+  CheckBox,
+  EllipsisText,
+  InputItem,
+  LinearGradient,
+  RadioButton,
+  ScreenTemplate,
+  Text,
+} from 'app/components';
+import { HEADER_HEIGHT } from 'app/components/Header';
+import { ImportWalletType, Wallet } from 'app/consts';
+import { HDSegwitP2SHArWallet, HDSegwitP2SHAirWallet } from 'app/legacy';
+import { createPin, createTxPassword, setIsTcAccepted } from 'app/state/authentication/actions';
+import { importWallet } from 'app/state/wallets/actions';
+import { getStatusBarHeight, typography, palette } from 'app/styles';
+
+const i18n = require('../../loc');
+
+const darkSlytherinColor = '#1a472a';
+const lightSlytherinColor = '#2a623d';
+
+const bloodColor = '#8A0303';
+
+const chamberOfSecretsGradient = {
+  colors: [darkSlytherinColor, lightSlytherinColor],
+  start: { x: 1, y: 0 },
+  end: { x: 1, y: 1 },
+};
+
+interface Props {
+  createPin: (value: string, { onSuccess }?: { onSuccess?: () => void }) => void;
+  createTxPassword: (value: string, { onSuccess }?: { onSuccess?: () => void }) => void;
+  setIsTcAccepted: (value: boolean) => void;
+  importWallet: (wallet: Wallet, { onSuccess }?: { onSuccess?: () => void }) => void;
+  onButtonPress: () => void;
+}
+
+const ChamberOfSecrets = (props: Props) => {
+  const { createPin, createTxPassword, setIsTcAccepted, onButtonPress } = props;
+
+  const [addWallet, setAddWallet] = useState(false);
+  const [walletOptions, setWalletOptions] = useState({
+    type: 'Standard',
+    name: 'BAD MOTHER FUCKER',
+    seedPhrase:
+      'innocent surge canoe iron nasty dinner aspect bar brand couch input embrace piano example panic champion aerobic debris search fly repeat sand nuclear tuition',
+  });
+
+  const defaultPin = '1234';
```

Comment:
I get the point, but I don't like when a component depends on external variables like this. So, since now `credentials` is an optional property with default value. Please have a look and judge if it makes more sense now